### ISSUE TITLE
Optional variable that allows collapsing the recently expanded cell, when any other cell is expanded

### ISSUE THREAD

**Issue by okhanokbay** [okhanokbay/ExpyTableView#9]

**Comment by vaibhav-varshaaweblabs** [comment]
@okhanokbay Here's an example

I have my first section expanded. 
Now, when I click on second section, the first section should collapse and second section should expand.

how to achieve this?

**Comment by okhanokbay** [comment]
@vaibhav-varshaaweblabs  Until it will be implemented in v0.5.0, you can achieve this by using `expyStateChangeForSection:` delegate method as written below: (Don't forget to set delegate)

 ```swift
extension ViewController: ExpyTableViewDelegate {
 //OPTIONAL DELEGATE METHOD, receives callbacks when a section will expand, will collapse, did expand, did collapse. A unified method.
  func tableView(_ tableView: ExpyTableView, expyState state: ExpyState, changeForSection section: Int) {

    switch state {
    case .willExpand:
    //call this method here OR in didExpand state below
    //collapseLatestExpandedSection(section)
     print("WILL EXPAND")

    case .didExpand:
    //call this function here OR in willExpand state above
    //collapseLatestExpandedSection(section)
    print("DID EXPAND")
    }
  } 
```

It is up to you to place your logic in willExpand or didExpand state, try and decide the best that fits your condition.

For example if you write a function named  
```swift 
func collapseLatestExpandedSection(section: Int) 
```
Store the latest expanded section in a variable like below.
Then the implementation could be something like this:

 ```swift
var latestExpandedSection: Int?

func collapseLatestExpandedSection(section: Int) {
    guard let latestExpandedSection = latestExpandedSection else {
        latestExpandedSection = section
        return
    }
    guard section != latestExpandedSection else { return }
    //ExpandableTableView is your table view instance.
    expandableTableView.collapse(latestExpandedSection)
    latestExpandedSection = section
}
 ```

You can call this function from delegate method and it should work. Please tell me if it doesn't work :)

**Comment by vaibhav-varshaaweblabs** [comment]
@okhanokbay Can we get the state of the cell on click of it?

**Comment by okhanokbay** [comment]
@vaibhav-varshaaweblabs You may want to look at `ExpyTableViewHeaderCell` protocol. Conform to it in your table view cell subclass then you get the state information updates.

**Comment by shahdhiren** [comment]
Is this feature now available in latest 1.0 in cocoapods?
